ci/WORKSPACE.filter.example

```diff
@@ -1,12 +1,12 @@
-workspace(name = "envoy")
+workspace(name = "envoy_filter_example")
```

Comment:
The ASAN/TSAN test failures are real, CI run ASAN/TSAN from envoy-filter-example though our test setup depends on TEST_WORKSPACE env. Before this change it is identical so wasn't a problem though now it is.

Reply:
Addressed by splitting into separate 'bazel test' invocations, so the @envoy//... tests are run with TEST_WORKSPACE=envoy, while the @envoy-filter-example//... tests are run with TEST_WORKSPACE=envoy-filter-example.  All CI tests pass now.

@lizan PTAL again, thanks!